protos/sigstore_verification.proto

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.verification.v1;
+
+import "sigstore_common.proto";
+import "sigstore_trustroot.proto";
+import "sigstore_bundle.proto";
+
+option go_package = "github.com/sigstore/proto/verification/v1";
+option java_package = "dev.sigstore.proto.verification.v1";
+option java_multiple_files = true;
+option java_outer_classname = "VerificationProto";
+
+// The identity of a signer.
+message Identity {
+        // The X.509v3 issuer extension (OID 1.3.6.1.4.1.57264.1.1)
+        string issuer = 1;
+        // A regular expression describing the expected value for the SAN.
+        string subject_alternative_name = 2;
+}
+
+message TransparencyLogs {
+        // The set of transparency logs to use during bundle verification.
+        repeated dev.sigstore.trustroot.v1.TransparencyLogInstance instances = 1;
+        // The minimim number of logs the entry must appear on.
+        // Setting this value to zero disables verification.
```

Comment:
"disables verification" means: always passes? or always fails?

Is that a dangerous behavior? I worry you could corrupt a `TransparencyLogs` message and it'd default to zero

Reply:
Disables as in always pass.

I wouldn't think there is a risk of corrupting a message, as the flow I envision is something like this:

1. The verifier instance is started.
2. The verifier gathers all crypto materials into a `TrustedRoot` structure.
3. Upon verifying a bundle (artifact), the verifier filters out what material that is needed from the `TrustedRoot`.
4. Based on configured polices (such as `thresholds`) and the filtering from step 3, the verifier prepares an `Input` structure.
5. Verifier calls verify with bundle and the `Input`.

Note that this is not something that exists today, this is the desired implementation steps after the discussions we had last week. So even if there is a "client exposed high level verify method", the implementation behind it is layered, to make sure we have good separation of concerns, to be able to have methods that are easy to test, validate and implement. 

This architecture is yet to be documented somewhere. I kind of eluded to it in some of the comments only.

Reply:
I think that flow makes sense.

I still worry that you might accidentally instantiate a `TransparencyLogs` proto without setting `threshold`, which makes it default to zero, which would be bad.

Reply:
Good point on default value, let me see if I can rearrange this a bit to make the exclusion of verification an explicit action.